src/main/kotlin/exercise2/task1/FindPairOfHighestSum.kt

```diff
@@ -18,7 +18,60 @@ import org.jetbrains.exercise2.task3.findPairWithBiggestDifference
  */
 
 internal fun List<Int>.findHighestSumPair(): Pair<Int, Int> {
-    TODO("Implement me!!")
+
+
+//    for ( i in 0  ..this.size - 1){
+//        if(this[i] >= first){
+//            first = this[i]
+//        }
+//    }
+//
+//    for ( i in 0  .. this.size - 1){
+//        if(this[i] > second && this[i] < first){
+//            second = this[i]
+//        }
+//    }
+
+//    var first = this[0]
+//    var second = this[0]
+//    println("Velicina " + this.size)
+//    for (i in 0 ..< this.size -1){
+//        if (this[i+1] >= first ) first = this[i+1]
+//        if ((this[i] > second) && (this[i] < first)) second = this[i]
+//    }
+//    if ((this[lastIndex] > second) && (this[lastIndex] < first)) second = this[lastIndex]
+//    println("Prvi: " + first)
+//    println("Drugi: " + second)
+
+
+//    Prlazak kroz dve petlje
+//    var first = this[1]
+//    var second = this[0]
+//    for ( i in indices){
+//        if(this[i] >= first){
+//            first = this[i]
+//        }
+//    }
+//
+//    for ( i in indices){
+//        if((this[i] > second) && (this[i] < first)){
+//            second = this[i]
+//        }
+//    }
+
+//    Prolazak kroz jednu petlju
+    var first = this[0]
+    var second = this[0]
+    for (i in 0 ..< this.size -1){
```

Comment:
If we run your implementation for the input `listOf(3, 1, 2)`, we would get the incorrect result `Pair(3, 3)`. 
Are the tests green for you?

nit: `this.size -1` can be written as `this.lastIndex`

Reply:
The test data for this method seems incomplete, and your implementation has a false positive in the tests, meaning the tests are green, but the actual implementation is incorrect.  

In the above example for the input `listOf(3, 1, 2)` the expected output should be `Pair(3, 2)`, and you implementation returns `Pair(3, 3)` which is incorrect. This is because the test didn't cover the case where the highest number in the list is at the first index. 

I'm very sorry for the inconvenience with incomplete test data.

Please add a missing case to the test class and fix the bug in your implementation.